editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -670,9 +670,8 @@ export const previewStore: StateCreator<
   },
 
   isFinalCard: () => {
-    // Temporarily always returns false until upcomingCardIds is optimised
-    // OSL Slack explanation: https://bit.ly/3x38IRY
-    return false;
+    const { upcomingCardIds } = get();
+    return upcomingCardIds().length === 1;
```

Comment:
This one's been a long time coming ([original commit](https://github.com/theopensystemslab/planx-new/commit/f67560d483954829d8e4bd79b0a22a646bb8852d#diff-d93e4413e0cb16ffeec24f6df1499fd4f24074ba6837ece6c782bb90053e0f37R638)) & more promising now since `upcomingCardIds` has been recently refactored to only be responsible for navigation and no longer optimistic auto-answering

Reply:
Very happy to see this commented back in 👍